test/extended/build_context_dir_test.go

```diff
@@ -0,0 +1,68 @@
+package extended
+
+import (
+	"testing"
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/test/util"
+)
+
+func init() {
+	util.RequireDocker()
+	util.RequireServer()
+}
+
+func TestBuildContextDir(t *testing.T) {
+	build := mockBuildWithContextDir()
+	client := util.GetClient()
+	if _, err := client.Builds("contextdir").Create(build); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	retries := 0
+
+	for {
+		time.Sleep(time.Duration(1 * time.Second))
+		current, err := client.Builds("contextdir").Get("sample")
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+		switch current.Status {
+		case buildapi.BuildStatusFailed, buildapi.BuildStatusError:
+			t.Fatalf("Unexpected failure of build: %v", current)
+		case buildapi.BuildStatusComplete:
```

Comment:
the build completing doesn't prove that you actually built the app using the correct contextDir or that the output was valid/correct.  needs more validation imho.